executor/src/witgen/machines/mod.rs

```diff
@@ -151,6 +150,67 @@ impl<'a, T: FieldElement> Machine<'a, T> for KnownMachine<'a, T> {
     }
 }
 
+#[derive(Clone, Copy)]
+pub struct ConnectingIdentity<'a, T> {
```

Comment:
Maybe one line of documentation?

Reply:
done

Reply:
Hm, not sure I would agree to this. A polynomial identity can also cross machine boundaries (depending on the definition, it would be the same machine, but still...). Shouldn't we be more explicit about the kind? Or the fact that it has two parts that do not need to match on the same row or something like that if you want to be abstract?

Reply:
I would just be explicit and say, lookup or permutation.